src/test/java/com/puppycrawl/tools/checkstyle/grammar/java21/Java21AstRegressionTest.java

```diff
@@ -72,4 +79,40 @@ public void testUnnamedVariableSwitch() throws Exception {
                         "InputUnnamedVariableSwitch.java"));
     }
 
+    @Test
+    public void testTextBlockConsecutiveEscapes() throws Exception {
+        verifyAst(
+                getNonCompilablePath(
+                        "ExpectedTextBlockConsecutiveEscapes.txt"),
+                getNonCompilablePath(
+                        "InputTextBlockConsecutiveEscapes.java"));
+    }
+
+    @Test
+    public void testTextBlockParsingFail() throws Exception {
```

Comment:
Unusual test case, but important to prevent regressions. We need to make sure that we only consume legal escapes in text blocks, and don't unintentionally parse something that should be an escape as regular text.

Reply:
Good idea, done